atomic_reactor/pulp_util.py

```diff
@@ -224,3 +245,61 @@ def get_registry_hostname(self):
 
     def get_pulp_instance(self):
         return self.pulp_instance
+
+
+class LockedPulpRepository(object):
```

Comment:
I think it makes more sense to pack a Lock argument to PulpHandler.update_repo that tells it to perform the locking, instead of requiring users of PulpHandler to do it themselves.

Reply:
Fair point, will take a look at doing it this way.

Reply:
I don't want to meddle in your PRs, but consider implementing some watchdog mechanism to avoid stalled locked repos. Something like: Ah, this lock repo was created 10 hours ago, there's not way it's still relevant to any running process, so lets remove it and lock it again for new process.